### PR TITLE
Add license check

### PR DESCRIPTION
**Pull request by yakiduck**
Add license check for CPU_flavor and Memory_flavor of node
and combine cluster status reasons

### CODE REVIEW

File: pkg/license/license.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+const (
+	LicenseKey = "license"
+)
+
+const (
+	limitsUnlimited = "Unlimited"
+)
+
+type License struct {
```
Comment (yylt) [inline pkg/license/license.go:32]: should give example about license

File: pkg/license/license.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+			current := capacity.Cpu().Value()
+			if current > limit {
+				isIllegal = true
+				clusterStatusReason = append(clusterStatusReason, fmt.Sprintf("Node: %s CPU Flavor: %d exceeds license limit", node.Name, current))
+			}
+		})
+		// Check each node Memory_flavor is illegal
+		check(l.MemoryFlavor, func(limit int64) {
+			current := capacity.Memory().Value()
+			if current > limit<<20 {
```
Comment (yylt) [inline pkg/license/license.go:87]: should comments,  Memory must be number, and its unit is MiB!
but it is not reasonable

File: pkg/license/license.go (hunk 1 above)
Comment (yakiduck) [reply pkg/license/license.go:32]: Done, and it is alse available in license_test.go

File: pkg/license/license.go (hunk 2 above)
Comment (yakiduck) [reply pkg/license/license.go:87]: Done, comment in License struct

**Pull request merged**